crates/compute/src/cpu/layer.rs

```diff
@@ -7,14 +7,72 @@ use binius_field::{
 };
 
 use super::{memory::CpuMemory, tower_macro::each_tower_subfield};
-use crate::layer::{ComputeLayer, Error, FSlice, FSliceMut};
+use crate::{
+	layer::{ComputeLayer, Error, FSlice, FSliceMut},
+	memory::FSliceWithTowerLevel,
+};
 
 #[derive(Debug)]
 pub struct CpuExecutor;
 
 #[derive(Debug, Default)]
 pub struct CpuLayer<F: TowerFamily>(PhantomData<F>);
 
+/// evals is treated as a matrix with `1 << log_query_size` rows and each column is dot-producted
+/// with the corresponding query element. The result is written to the `output` slice of values.
+/// The evals slice may be any field extension defined by the tower family T.
+fn compute_left_fold<EvalType: TowerField, T: TowerFamily>(
```

Comment:
nit: Move function to follow the `impl` block below. It will make the file easier to read.

In general, impl blocks should usually immediately follow `struct` definitions.